data-reconciliation-app/docs/adr/02-data-schema-strategy.md

```diff
@@ -0,0 +1,127 @@
+# Data Schema Strategy
+
+## Status
+
+Decided
+
+## Context
+
+We need to build an API that ingests member data into our system. Once ingested, we also need to store this data in the K-V store. Members will decide on a data schema which will be used for ingesting, data conciliation and reporting.
+
+There are multiple strategies for data schema definition and usage. In this document, we lay out our decision and some other alternatives considered.
+
+## Alternatives
+
+### Option1: Fully Flexible Data Schema
+
+In this strategy, members define their data schema and submit it at each request as part of data ingestion. API will take the schema and map it to internal universal schema before storing in the K-V store. When a member requests data reconciliation report, API will generate the report using the internal universal schema. This will result in member to understand the schema of the report and possibly convert it back to their custom format to use at their side.
```

Comment:
```suggestion
In this strategy, members define their data schema and submit it as part of data ingestion. The API will take the schema and map it to an internal universal schema before storing in the K-V store. When a member requests a data reconciliation report, the API will generate the report using the internal universal schema. This will result in member to understand the schema of the report and possibly convert it back to their custom format to use at their side.
```

Reply:
Also that last sentence, I get the general meaning, but it's a bit cumbersome - could you just re-read it?